yocto-update-strategies.md

```diff
@@ -0,0 +1,651 @@
+compatibility with the Yocto framework.
+
+Tests are essential for validating the functionality of the project during
+updates. It is crucial to have a comprehensive set of tests that cover all
+custom functionalities within the project. Automating tests, if possible,
+improves the frequency and reliability of testing, enabling faster and more
+efficient updates.
+
+Implementing CI/CD processes for build and tests is highly beneficial in
+updating Yocto-based projects. CI/CD automation helps streamline the building
+and testing of the project, reducing manual efforts and saving time. By
+automating these processes, we ensure consistent and reliable results,
+maintaining the integrity of the project throughout the update process.
+
+---
+# Project components
+
+.center[<img src="img/compo-life.svg" width="550px">]
+
+???
+
+Let's explore how the different project components align with the stages of the
+project lifecycle we discussed earlier. During the initiation phase,
+architecture documentation is created, providing a foundational understanding of
+the system structure. This documentation becomes particularly valuable during
+updates, helping us identify areas that may require modifications or
+improvements.
+
+Moving into the planning phase, features documentation is developed, outlining
+the desired functionalities of the project. This documentation guides us in
+prioritizing updates based on project requirements and ensures that our updates
+align with the intended goals of the system.
+
+As we progress into the development phase, the build environment is established.
+It provides the necessary tools, configurations, and dependencies for building
+and updating the project. Maintaining an up-to-date build environment is crucial
+for efficient updates, ensuring compatibility with the latest Yocto releases.
+
+During the testing phase, comprehensive tests are designed and implemented.
+These tests validate the project's functionality and help us identify potential
+regressions that may arise during updates. By conducting thorough tests, we can
+ensure that the system remains stable and reliable after applying updates.
+
+In the deployment phase, CI/CD processes come into play. These automated
+workflows streamline the build and testing stages, making the deployment of
+updates more efficient and reproducible. CI/CD ensures that updates are
+integrated seamlessly into the system.
+
+Finally, all of these project components, including architecture documentation,
+features documentation, build environment, tests, and CI/CD, collectively
+contribute to ongoing maintenance and updates. They form a cohesive framework
+that supports the project throughout its lifecycle, enabling us to address
+security vulnerabilities, minimize technical debt, and integrate new
+functionalities as we maintain and update our Yocto-based projects.
+
+---
+# Update strategy
+
+* Remembering every project component during updates
```

Comment:
Definitely too much text on this slide
Ideally if the bullet point is only one line long, maximum of two lines

we need some short phrases as bullet points, not necessarily the whole sentences

Reply:
I put here a little bit less text, fix https://github.com/3mdeb/conferences/pull/7/commits/47359d083f2e6c8919c2292127ced92fe98d1571